src/component/framework/conditional.js

```diff
@@ -27,3 +45,7 @@ export class Conditional {
   }
 
 }
+
+function isPromise(value) {
+  return (typeof value === 'object' && value.then);
```

Comment:
we do write esNext (or maybe typescript in future), and that has Promises, They are all over the place and if needed already added by polyfills to global. so, i think that concern is valid.